h/static/scripts/directives/thread.coffee

```diff
@@ -137,6 +142,12 @@ thread = [
         else
           attrs.$removeClass COLLAPSED_CLASS
 
+      scope.$watch $parse(attrs.thread), (thread) ->
```

Comment:
The watch is necessary because the computed value of the attribute
expression may change. Won't happen when we use the thread directive in a
repeat, since the element will be torn down whenever the thread might have
changed, but we shouldn't assume that here, unless we want to advertise
that the thread expression is fixed at link time and should not change.

As for the explicit digest, the render service is there to ensure large
conversations don't block rendering while they load. It does not execute
the callback in a root scope $apply so that the caller can do a more local,
and thereby faster, digest. A better interface might be like $timeout,
where the default is to use $apply but an optional argument can skip it.
On Jan 23, 2015 3:24 PM, "Nick Stenning" <notifications@github.com> wrote:

> In h/static/scripts/directives/thread.coffee
> <https://github.com/hypothesis/h/pull/1830#discussion_r23451880>:
>
> > @@ -137,6 +142,12 @@ thread = [
> >          else
> >            attrs.$removeClass COLLAPSED_CLASS
> >
> > +      scope.$watch $parse(attrs.thread), (thread) ->
>
> I'm going to merge this, as it fixes some bugs currently in prod. I'm
> still a bit dubious about this pattern, though, as it seems to me like
> explicitly having to call $digest like this implies we're using the scope
> wrong. At the very least, I'd like to see a comment explaining why this
> $watch is necessary.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hypothesis/h/pull/1830/files#r23451880>.
>